worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionData.java

```diff
@@ -59,20 +59,25 @@ class MapPartitionData implements MemoryManager.ReadBufferTargetChangeListener {
   private final BufferQueue bufferQueue = new BufferQueue();
   private AtomicBoolean bufferQueueInitialized = new AtomicBoolean(false);
   private MemoryManager memoryManager = MemoryManager.instance();
-  private Consumer<Long> recycleStream;
+  protected Consumer<Long> recycleStream;
   private int minReadBuffers;
   private int maxReadBuffers;
   private int minBuffersToTriggerRead;
   private AtomicBoolean hasReadingTask = new AtomicBoolean(false);
 
+  // if requireSubpartitionId is set to true, means that the buffer sent to client
+  // by the worker should be SubpartitionReadData rather than ReadData
+  protected boolean requireSubpartitionId;
```

Comment:
requireSubpartitionId is necessary only when using SegmentMapPartitionData

Reply:
Yes, we cloud move this field to `SegmentMapPartitionData`, updated.